recipes-bsp/u-boot/files/upstream/0027-iot2050-Add-support-for-flash-protect.patch

```diff
@@ -0,0 +1,32 @@
+From fbb415a6ad15195146ef1f2720032ba17555d38c Mon Sep 17 00:00:00 2001
+From: Su Baocheng <baocheng.su@siemens.com>
+Date: Wed, 20 Jan 2021 16:52:19 +0800
+Subject: [PATCH 27/28] iot2050: Add support for flash protect
+
+IOT2050 use a Winbond flash that support lock & unlock, but it was not
+enabled.
+
+This winbond flash is a similar one to STMICRO, and can reuse the lock
+unlock logic for STMICRO chips. So by adding CONFIG_SPI_FLASH_STMICRO
+to iot2050_defconfig, the flash protection could be supported.
+
```

Comment:
If there is a technical dependency between CONFIG_SPI_FLASH_WINBOND and CONFIG_SPI_FLASH_STMICRO now (or after patch 28), that needs to be expressed via Kconfig instead. If we only use STMICRO now, WINBOND must be dropped.

Reply:
The flash on our board is Winbond, but it follows the same logic of flash protection with ST Micro flash.

Some in-tree boards also does things like this.